src/flint/types/fmpq_mpoly.pyx

```diff
@@ -94,6 +94,22 @@ cdef class fmpq_mpoly_ctx(flint_mpoly_context):
         fmpq_mpoly_ctx_init(self.val, nvars, ordering_py_to_c(ordering))
         super().__init__(nvars, names)
 
+    def any_as_scalar(self, other):
+        if isinstance(other, int):
+            return any_as_fmpq(other)
+        elif typecheck(other, fmpz):
+            return any_as_fmpq(other)
+        elif typecheck(other, fmpq):
+            res = fmpq.__new__(fmpq)
+            fmpq_set((<fmpq>res).val, (<fmpq>other).val)
+            return res
```

Comment:
If it is an `fmpq` already can we not just return it?

Why does a copy need to be made?

Reply:
I thought it best to be consistent with the other `any_as_*` methods, but since they are immutable on a `python-flint` level perhaps we can just return the same object.

`any_as_fmpq` is just not used there because no conversion is required

```python
>>> from flint import *
>>> x = fmpz(1)
>>> y = any_as_fmpz(x)
>>> x is y
False
```

(With `any_as_fmpz` modified to be a `cpdef`)

Reply:
Maybe `any_as_fmpz` should be changed to not make a copy. I'm not sure that is compatible with how it is currently used though like whether the object returned is ever mutated.